src/app/shared/services/task/task.service.ts

```diff
@@ -17,14 +20,17 @@ export class TaskService {
 
   async init() {
     await this.getListOfTaskGroups();
-    this.evaluateHighlightedTaskGroup();
+    if (this.taskGroups.length) {
```

Comment:
nit(blocking)
Can you add `>0`? I know zero is falsy but still just feels a bit wrong to me.
Also I think no harm initialising both the taskGroups and taskGroupsHashmap variables (as `[]` and `{}`) respectively, just to tidy up in cases where issues happen during init